src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -0,0 +1,199 @@
+    }
+
+    /**
+     * Pass back the list of ReadFilter instances that were actually seen on the
+     * command line in the same order they were specified
+     */
+    @Override
+    public void getInstances(final Consumer<Collection<ReadFilter>> consumer) {
+        // add the instances in the order they were specified on the command line
+        final ArrayList<ReadFilter> filters = new ArrayList<>(readFilters.size());
+        readFilterNames.forEach(s -> filters.add(getReadFilterForName(s)));
+        consumer.accept(filters);
+    }
+
+    // Find the instance of the given read filter in the instance list.
+    private ReadFilter getReadFilterForName(final String filterName) {
+        ReadFilter rf = null;
+        for (Map.Entry<String, ReadFilter> entry : readFilters.entrySet()) {
+            if (entry.getKey().endsWith(filterName)) {
+                rf = entry.getValue();
+                break;
+            }
+        }
+        return rf;
+    }
+
+    // Return the allowable values for readFilterNames/disableReadFilter
+    @Override
+    public Set<String> getAllowedStringValues(final String longArgName) {
+        if (longArgName.equals(StandardArgumentDefinitions.READ_FILTER_LONG_NAME) ||
+                longArgName.equals(disableReadFilterArgName)) {
+            return readFilters.keySet();
+        }
+        throw new IllegalArgumentException("Allowed values request for unrecognized string argument: " + longArgName);
+    }
+
+    /**
+     * Validate the list of arguments and reduce the list of read filters to those
+     * actually seen on the command line. This is called by the command line parser
+     * after all arguments have been parsed.
+     */
+    @Override
+    public void validateArguments() {
+        // first, validate that no filter is disabled *and* enabled
+        disableFilters.forEach(
+            s -> {
+                if (readFilterNames.contains(s)) {
+                    throw new UserException.CommandLineException(
+                            "Read filter: " + s + " is both enabled and disabled");
+                }
+            });
+        readFilterNames.forEach(
+            s -> {
+                if (disableFilters.contains(s)) {
+                    throw new UserException.CommandLineException(
+                            "Read filter: " + s + " is both enabled and disabled");
+                }
+            });
+
+        // now validate that each filter specified is valid (has a corresponding instance)
+        Map<String, ReadFilter> requestedReadFilters = new HashMap<>();
+        readFilterNames.forEach(s -> {
+            ReadFilter trf = readFilters.get(s);
+            if (null == trf) {
+                throw new UserException.CommandLineException("Unrecognized read filter name: " + s);
+            }
+            else {
+                requestedReadFilters.put(s, trf);
+            }
+        });
+        readFilters = requestedReadFilters;
```

Comment:
I think all of this validation should be moved to a second pass that takes into account the tool-requested filters, as proposed above.